modules/core/rapid_context.js

```diff
@@ -1,34 +1,72 @@
 import { geoExtent } from '../geo';
 
 import toGeoJSON from '@mapbox/togeojson';
+import { dispatch as d3_dispatch } from 'd3-dispatch'; 
+import { utilRebind } from '../util';
 
-
-export function coreRapidContext(context) {
+export function coreRapidContext() {
     var rapidContext = {};
     rapidContext.version = '1.0.1';
+    var _isRect = undefined; 
+    var dispatch = d3_dispatch('task_extent_set');
+
+    function distinct (value, index, self) {
+        return self.indexOf(value) === index; 
+    }
 
     var taskExtent;
     rapidContext.setTaskExtentByGpxData = function(gpxData) {
         var dom = (new DOMParser()).parseFromString(gpxData, 'text/xml');
         var gj = toGeoJSON.gpx(dom);
+        var uniqueLats = []; 
+        var uniqueLngs = []; 
         if (gj.type === 'FeatureCollection') {
             var minlat, minlon, maxlat, maxlon;
+            //Calculate task extent. 
             gj.features.forEach(function(f) {
                 if (f.geometry.type === 'Point') {
                     var lon = f.geometry.coordinates[0];
                     var lat = f.geometry.coordinates[1];
                     if (minlat === undefined || lat < minlat) minlat = lat;
                     if (minlon === undefined || lon < minlon) minlon = lon;
                     if (maxlat === undefined || lat > maxlat) maxlat = lat;
-                    if (maxlon === undefined || lon > maxlon) maxlon = lon;
+                    if (maxlon === undefined || lon > maxlon) maxlon = lon;                    
+                }
+    
+                if (f.geometry.type === 'LineString') {
+                    var lats = f.geometry.coordinates.map(function(f) {return f[0];});
+                    var lngs = f.geometry.coordinates.map(function(f) {return f[1];});
+
+                    var uniqueLatCount = lats.filter(distinct).length; 
+                    var uniqueLngCount = lngs.filter(distinct).length; 
+
+                    //If there are only two unique lngs and two unique lats in the geometry of the 
+                    //task area, congrats- we have a rectangle! 
+                    if (uniqueLatCount === 2 && uniqueLngCount === 2) {
+                        _isRect = true; 
```

Comment:
besides this condition, we should also make sure that along each unique lat/lon value, there are at least 2 points. e.g. one counter-case could be a linestring containing only two points: `uniqueLatCount === 2 && uniqueLngCount === 2` will be true but it's not a rectangle. Using a two dicts keyed on lat/lon values might be the easies way to help with that.